sql/session.go

```diff
@@ -139,6 +139,15 @@ type Session interface {
 	GetCharacterSetResults() CharacterSetID
 	// GetCollation returns the collation for this session (defined by the system variable `collation_connection`).
 	GetCollation() CollationID
+	// GetPrivilegeSetCounter returns the counter associated with this privilege set for this session. A value of zero
+	// indicates that there is no cached privilege set.
+	GetPrivilegeSetCounter() uint64
```

Comment:
I would eliminate this method and return the counter from the signature of GetPrivilegeSet instead (`GetPrivilegeSet (PrivilegeSet, uint64)`). They're pretty coupled.

Also mention in these docs that these methods are for internal use only and not meant to be called / implemented by integators.